pkg/operator/api/api.go

```diff
@@ -224,7 +240,23 @@ func (a *apiServer) ListComponents(ctx context.Context, in *operatorv1pb.ListCom
 		Components: [][]byte{},
 	}
 	for i := range components.Items {
+		// By default assume that it is not a component for control plane service.
+		controlPlaneComp := false
 		c := components.Items[i] // Make a copy since we will refer to this as a reference in this loop.
+		// If the component is in the same namespace as the operator AND has a scope defined with a dynamic control plane service name, then it is a control plane component.
+		if c.ObjectMeta.Namespace == security.CurrentNamespace() {
+			for s := range c.Scopes {
+				scope := c.Scopes[s]
+				if utils.IsControlPlaneService(scope) {
+					controlPlaneComp = true
+					break
+				}
+			}
+		}
+		if (!controlPlaneServiceReq && controlPlaneComp) || (controlPlaneServiceReq && !controlPlaneComp) {
+			continue
+		}
```

Comment:
Why do we want it so that a Component which has a scope that defines any control plane service will not also be sent to a non control plane application, rather than continuing to filter based solely on the service identity?

Reply:
Yeah, that's the ask. A request is identified whether to be from a CP or non-CP.
And, components are identified whether meant for CP or non-CP.
Only matching components are sent across.